src/components/tag/bl-tag.ts

```diff
@@ -0,0 +1,95 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+import { event, EventDispatcher } from "../../utilities/event";
+import "../button/bl-button";
+import "../icon/bl-icon";
+import { BaklavaIcon } from "../icon/icon-list";
+import style from "./bl-tag.css";
+
+export type TagSize = "small" | "medium" | "large";
+type TagVariant = "selectable" | "removeable";
+
+/**
+ * @tag bl-tag
+ * @summary Baklava Tag component
+ */
+
+@customElement("bl-tag")
+export default class BlTag extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query(".remove-button") removeButton!: HTMLButtonElement;
+
+  /**
+   * Sets the tag size
+   */
+  @property({ type: String, reflect: true })
```

Comment:
Can we use TagSize and TagVariant as types on decorators instead?

Reply:
we can't. TagVariant and TagSize are TypeScript types, you should pass JavaScript definitions here.